pkg/arvo/sys/arvo.hoon

```diff
@@ -968,289 +1017,297 @@
         bud=vase                                        ::  %zuse
         vanes=(list [label=@tas =vane])                 ::  modules
     ==                                                  ::
-=<  ::  Arvo structural interface
-    ::
-    |%
-    ++  come  |=  [@ @ @ (list ovum) vise pone]         ::   4
-              ^-  [(list ovum) _+>]
-              ~&  %hoon-come
-              =^  rey  +>+  (^come +<)
-              [rey +>.$]
-    ::
-    ++  load  |=  [@ @ @ (list ovum) vase pane]         ::  10
-              ^-  [(list ovum) _+>]
-              ~&  %hoon-load
-              =^  rey  +>+  (^load +<)
-              [rey +>.$]
-    ::
-    ++  peek  |=  *                                     ::  46
-              =/  rob  (^peek ;;([@da path] +<))
-              ?~  rob  ~
-              ?~  u.rob  ~
-              [~ u.u.rob]
-    ::
-    ++  poke  |=  *                                     ::  47
-              =>  .(+< ;;([now=@da ovo=ovum] +<))
-              =^  ova  +>+.$  (^poke now ovo)
-              =|  out=(list ovum)
-              |-  ^-  [(list ovum) *]
-              ?~  ova
-                [(flop out) +>.^$]
-              ::  upgrade the kernel
-              ::
-              ?:  ?=(%lyra -.q.i.ova)
-                %+  fall
-                  (vega now t.ova ;;([@ @] +.q.i.ova))
-                [~ +>.^$]
-              ::  iterate over effects, handling those on arvo proper
-              ::  and passing the rest through as output
-              ::
-              =^  vov  +>+.^$  (feck now i.ova)
-              =?  out  ?=(^ vov)  [+.vov out]
-              $(ova t.ova)
-    ::
-    ++  wish  |=(* (^wish ;;(@ta +<)))             ::  22
-    --
-::  Arvo implementation core
+::  produce an interface that vere understands
+::
+=<  arvo
+::  soul: arvo implementation core
 ::
 |%
-++  come                                                ::  load incompatible
-  |=  [who=ship now=@da yen=@ ova=(list ovum) dub=vise nyf=pone]
-  ^+  [ova +>]
+++  soul  .
+::  arvo: structural interface core
+::
+++  arvo
+  |%
+  ++  come  |=  [@ @ @ pram vise pone]         ::   4
+            (come:soul +<)
+  ::
+  ++  load  |=  [@ @ @ pram vase pane]         ::  10
+            (load:soul +<)
+  ::
+  ++  peek  |=  *                                     ::  46
+            =/  rob  (^peek ;;([@da path] +<))
+            ?~  rob  ~
+            ?~  u.rob  ~
+            [~ u.u.rob]
+  ::
+  ++  poke  |=  *                                     ::  47
+            =>  .(+< ;;([now=@da ovo=ovum] +<))
+            (poke:soul now ovo)
+  ::
+  ++  wish  |=(* (^wish ;;(@ta +<)))             ::  22
+  --
+::  $pram: upgradeable state
+::  +dram: upgrade $pram, compatible
+::
++$  pram
+  $~  [%345 nub=*debt:le:part]
+  $%  [%345 nub=debt:le:part]
```

Comment:
Kelvin versions are great. I'm a uuuge fan of kelvin versions. But this is not a valid kelvin version, as it violates the stack invariants (see `%zuse`). I humbly request that this be replaced with a date constant, until such time as we can normalize the versioning across the whole kernel.